autogtp/Game.h

```diff
@@ -0,0 +1,63 @@
+/*
+    This file is part of Leela Zero.
+    Copyright (C) 2017 Marco Calignano
+
+    Leela Zero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Leela Zero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef GAME_H
+#define GAME_H
+#include <QProcess>
+#include <QTextStream>
+
+class Game : QProcess {
+public:
+    Game(QString weights, QTextStream *out);
+    ~Game(){}
+    void gameStart();
+    void move();
+    bool waitForMove() {return waitReady();}
+    bool readMove();
+    bool nextMove();
+    bool getScore();
+    bool writeSgf();
+    bool dumpTraining();
+    void gameQuit();
+
+private:
+    QTextStream *output;
+    QString cmdLine;
+    QString timeSettings;
+    QString moveDone;
```

Comment:
Doesn't need to be a member, only used in one function.

Reply:
cmdLine and timeSetting are member because maybe we can tune them in the future via the command parameter for autogtp.

Reply:
You would construct cmdLine off of some other options (like a list of GPUs), so you'd put those extra options as the members then. timeSetting = infinite time. I doubt we'll change this.